modules/bootstrapped/test/src/smithy4s/schema/FieldSpec.scala

```diff
@@ -52,4 +52,26 @@ final class FieldSpec extends FunSuite {
     assertEquals(result.toList, List(Some("test2"), None))
   }
 
+  test(
```

Comment:
Thank you for the additional tests. I'd like to see one for the following additional cases : 

```scala
val requiredNullable = string.nullable.required[Foo]("a", _.a)
```

and

```scala
case class Bar(
  a: Option[Nullable[String]]
)  

val optionalNullable = string.nullable.optional[Bar]("a", _.a)
```

Reply:
Added! I also took the liberty of adding one for `string.optional`, because we didn't have one before and I wanted to make sure that the behaviour I was getting on `string.nullable.optional` was the same as the one on `string.optional`.